sui_types/src/error.rs

```diff
@@ -30,6 +30,16 @@ macro_rules! fp_ensure {
 }
 pub(crate) use fp_ensure;
 
+#[macro_export]
+macro_rules! exit_main {
+    ($result:expr) => {
+        match $result {
+            Ok(_) => (),
+            Err(_) => std::process::exit(1),
```

Comment:
Why not propagate the error if it exists (i.e., it was not panicked?)

Reply:
Ah forgot to print the error. Let me update.
Previously it would panic and print a stacktrace, which is not good from UX perspective.